src/services/tx/tx-sender/dispatch.ts

```diff
@@ -333,9 +333,53 @@ export const dispatchTxRelay = async (
     txDispatch(TxEvent.RELAYING, { taskId, txId })
 
     // Monitor relay tx
-    waitForRelayedTx(taskId, txId)
+    waitForRelayedTx(taskId, [txId])
   } catch (error) {
     txDispatch(TxEvent.FAILED, { txId, error: error as Error })
     throw error
   }
 }
+
+export const dispatchBatchExecutionRelay = async (
+  txs: TransactionDetails[],
+  multiSendContract: MultiSendCallOnlyEthersContract,
+  multiSendTxData: string,
+  chainId: string,
+) => {
+  const to = multiSendContract.getAddress()
+  const data = multiSendContract.contract.interface.encodeFunctionData('multiSend', [multiSendTxData])
+  const groupKey = multiSendTxData
+
+  try {
+    const relayResponse = await sponsoredCall({
+      chainId,
+      to,
+      data,
+    })
+    const taskId = relayResponse.taskId
+
+    if (!taskId) {
+      throw new Error('Transaction could not be relayed')
+    }
```

Comment:
I believe `sponsoredCall` can already throw so we would never get to line 361. Do we still need that check?